openmls/src/framing/public_message_in.rs

```diff
@@ -0,0 +1,325 @@
+//! # PublicMessageIn
+//!
+//! A PublicMessageIn is a framing structure for MLS messages. It can contain
+//! Proposals, Commits and application messages.
+
+use crate::{error::LibraryError, group::errors::ValidationError, versions::ProtocolVersion};
+
+use super::{
+    mls_auth_content_in::{
+        AuthenticatedContentIn, FramedContentAuthDataIn, VerifiableAuthenticatedContentIn,
+    },
+    mls_content_in::{AuthenticatedContentTbmIn, FramedContentIn, FramedContentTbsIn},
+    *,
+};
+
+use openmls_traits::OpenMlsCryptoProvider;
+use std::{
+    convert::TryFrom,
+    io::{Read, Write},
+};
+use tls_codec::{
+    Deserialize as TlsDeserializeTrait, Serialize as TlsSerializeTrait, TlsDeserialize,
+    TlsSerialize, TlsSize,
+};
+
+/// Wrapper around a `Mac` used for type safety.
+#[derive(
+    Debug, PartialEq, Clone, Serialize, Deserialize, TlsSerialize, TlsDeserialize, TlsSize,
+)]
+pub(crate) struct MembershipTagIn(pub(crate) Mac);
```

Comment:
Do we have to have an -In type for MembershupTag? It doesn't seem to be used anywhere as far as I can tell.